# <center>2213645 刘嘉硕 第四次回归上机作业</center>

# 对A,B,C三家工厂生产的同一型号的电池比较其质量, 从各厂的产品中随机抽取6只电池，得到测试后的数据

## (1) 将数据表成方差分析模型的形式

$$
\begin{cases}
y_{ij} = \mu_i +e_{ij},\\
e_{ij} \sim N(0,\sigma^2)
\end{cases} \hspace{2em}
i = 1, \cdots, 3 \hspace{1em}
j = 1, \cdots, 6
$$
其中$\mu_i$是第$i$个工厂的均值，$e_{ij}$是第$i$个工厂第$j$个电池相应的实验误差

## (2) 在显著性水平α=0.05下检验三厂生产的电池的平均寿命有无显著影响? 列出方差分析表

In [1]:
data <- data.frame(
  BatteryLife = c(40, 48, 38, 42, 45, 46, 26, 34, 30, 28, 32, 33, 39, 40, 48, 50, 50, 52),
  Factory = factor(rep(c("A", "B", "C"), each = 6))
)
data

BatteryLife,Factory
<dbl>,<fct>
40,A
48,A
38,A
42,A
45,A
46,A
26,B
34,B
30,B


In [2]:
anova_result <- aov(BatteryLife ~ Factory, data = data)
summary(anova_result)

            Df Sum Sq Mean Sq F value   Pr(>F)    
Factory      2  855.1   427.6   23.25 2.54e-05 ***
Residuals   15  275.8    18.4                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

方差分析表为：

| 方差来源| 平方和 | 自由度 | 均方 | F比 |
|---------|---------|---------|---------|---------|
| 工厂    | 855.51   | 2   | 427.6 | 23.239  |
| 误差    | 275.8   | 15   | 18.4 | |
| 总和    | 1130.9   |  17  |     |  |

p值为$2.54 \times 10^{-5}$，小于α的值，所以认为在犯错误的概率不大于5%的情况下，拒绝三个工厂的平均寿命无显著差异的零假设，即认为三个工厂生产的电池的平均寿命有显著差异

## (3) 记$\mu_A$,$\mu_B$和$\mu_C$分别为三厂生产的电池的平均寿命, 写出均值之差$\mu_A-\mu_B$, $\mu_A-\mu_C$, $\mu_B-\mu_C$的95%的同时置信区间

In [3]:
# 分组数据
group_A <- data$BatteryLife[data$Factory == "A"]
group_B <- data$BatteryLife[data$Factory == "B"]
group_C <- data$BatteryLife[data$Factory == "C"]

# 计算均值
mean_A <- mean(group_A)
mean_B <- mean(group_B)
mean_C <- mean(group_C)

# 从方差分析表可以得到σ的估计值
sd <- sqrt(18.4)

n <- 6
alpha <- 0.05
t <- qt(1-alpha/6, 15)
CI_AB_upper <- (mean_A - mean_B) + t * sd * sqrt(1/n + 1/n)
CI_AB_lower <- (mean_A - mean_B) - t * sd * sqrt(1/n + 1/n)
CI_AC_upper <- (mean_A - mean_C) + t * sd * sqrt(1/n + 1/n)
CI_AC_lower <- (mean_A - mean_C) - t * sd * sqrt(1/n + 1/n)
CI_BC_upper <- (mean_B - mean_C) + t * sd * sqrt(1/n + 1/n)
CI_BC_lower <- (mean_B - mean_C) - t * sd * sqrt(1/n + 1/n)
cat("μA - μB的置信区间: [", CI_AB_lower, ", ", CI_AB_upper, "]\n")
cat("μA - μC的置信区间: [", CI_AC_lower, ", ", CI_AC_upper, "]\n")
cat("μB - μC的置信区间: [", CI_BC_lower, ", ", CI_BC_upper, "]\n")

μA - μB的置信区间: [ 5.995468 ,  19.33786 ]
μA - μC的置信区间: [ -10.00453 ,  3.337865 ]
μB - μC的置信区间: [ -22.6712 ,  -9.328802 ]
